diff --git a/home/tim/mpich/install/include/mpi.h b/home/tim/ancor-programm-traits/sample/pmpi_wrapper_test/mpi.h
index 1af5141..1fa230e 100644
--- a/home/tim/mpich/install/include/mpi.h
+++ b/home/tim/ancor-programm-traits/sample/pmpi_wrapper_test/mpi.h
@@ -1,3 +1,5 @@
+// this it the mpi.h from MPICH
+// only the thing about the matching wildcard usage is changed
 /*
  * Copyright (C) by Argonne National Laboratory
  *     See COPYRIGHT in top-level directory
@@ -500,9 +502,24 @@ extern int * const MPI_UNWEIGHTED MPICH_API_PUBLIC;
 extern int * const MPI_WEIGHTS_EMPTY MPICH_API_PUBLIC;
 
 #define MPI_PROC_NULL   (-1)
-#define MPI_ANY_SOURCE 	(-2)
 #define MPI_ROOT        (-3)
+
+#ifndef USE_WILDCARD
+//for ease of development: fully qualified header to our marker library
+//TODO changeme
+#include "/home/tim/ancor-programm-traits/markers.h"
+    //NO wildcards allowed
+#define MPI_ANY_SOURCE (-2) static_assert(0,"If you want to use matching wildcards: define USE_WILDCARD before including MPI header or compile with -DUSE_WILDCARD");
+#define MPI_ANY_TAG (-1) static_assert(0,"If you want to use matching wildcards: define USE_WILDCARD before including MPI header or compile with -DUSE_WILDCARD");
+
+marker(no_wildcard)
+    #else
+// normal mode with wildcards enabled
+#define MPI_ANY_SOURCE 	(-2)
 #define MPI_ANY_TAG     (-1)
+    #endif
+
+
 
 #define MPI_BOTTOM      (void *)0
 #define MPI_IN_PLACE  (void *) -1
