--- mpi.h.in.orig	2025-01-29 16:20:45.588477339 +0100
+++ mpi.h.in	2025-01-29 16:29:27.601584461 +0100
@@ -1,3 +1,4 @@
+// patched to include the marker integer for the absence of wildcards
 /*
  * Copyright (C) by Argonne National Laboratory
  *     See COPYRIGHT in top-level directory
@@ -500,9 +501,28 @@
 extern int * const MPI_WEIGHTS_EMPTY MPICH_API_PUBLIC;

 #define MPI_PROC_NULL   (-1)
-#define MPI_ANY_SOURCE 	(-2)
 #define MPI_ROOT        (-3)
+
+#ifndef USE_WILDCARD
+
+#ifdef __GNUC__  // GCC/Clang specific: prevent compiler form optimizing away the marker
+#define ATTRS __attribute__((used,visibility("default"))) inline
+#else
+#define ATTRS
+#endif
+
+#define MARKER_INTEGER_NAME_PREFIX marker_integer_
+// from markers.h:
+#define marker(x)  ATTRS int CONCAT(MARKER_INTEGER_NAME_PREFIX,x);
+//  needed for correct macro expansion
+#define CONCAT(x,y) CONCAT2(x,y)
+#define CONCAT2(x,y) x##y
+
+//NO wildcards allowed: compilation fails on usage
+#define MPI_ANY_SOURCE (-2) static_assert(0,"If you want to use matching wildcards: define USE_WILDCARD before including MPI header or compile with -DUSE_WILDCARD");
+#define MPI_ANY_TAG (-1) static_assert(0,"If you want to use matching wildcards: define USE_WILDCARD before including MPI header or compile with -DUSE_WILDCARD");
+// add the marker
+marker(no_wildcard)
+
+#else
+// normal mode with wildcards enabled
+#define MPI_ANY_SOURCE 	(-2)
 #define MPI_ANY_TAG     (-1)
+#endif
 
 #define MPI_BOTTOM      (void *)0
 #define MPI_IN_PLACE  (void *) -1
